core/pbcc_actions_api.c

```diff
@@ -59,8 +44,35 @@ enum pubnub_res pbcc_form_the_action_object(struct pbcc_context* pb,
                          actype);
         return PNR_INVALID_PARAMETERS;
     }
+
+    return pbcc_form_the_action_object_str(pb, obj_buffer, buffer_size, type_literal, val);
+}
+
+
+enum pubnub_res pbcc_form_the_action_object_str(struct pbcc_context* pb,
+                                            char* obj_buffer,
+                                            size_t buffer_size,
+                                            char const* action_type,
+                                            char const** val) {
+    char const* user_id = pbcc_user_id_get(pb);
+
+    PUBNUB_ASSERT_OPT(user_id != NULL);
+
+    if (NULL == user_id) {
+        PUBNUB_LOG_ERROR("pbcc_form_the_action_object(pbcc=%p) - user_id not set.\n", pb);
+        return PNR_INVALID_PARAMETERS;
+    }
+    if (('\"' != **val) || ('\"' != *(*val + pb_strnlen_s(*val, PUBNUB_MAX_OBJECT_LENGTH) - 1))) {
+        PUBNUB_LOG_ERROR("pbcc_form_the_action_object(pbcc=%p) - "
+                         "quotation marks on value ends are missing: "
+                         "value = %s\n",
+                         pb,
+                         *val);
+        return PNR_INVALID_PARAMETERS;
+    }
```

Comment:
If users now able to pass `string` as `type`, should we have the same check for it and ask users to enclose `type` into double quotes?

Reply:
But wait. 
previous implementation haven't include the quotes. So in this case they shouldn't be forced to do such a thing ;o